lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+
+
+Assuming `lastModified` is Date object you can set this header via [response object](http://hapijs.com/api#response-object) interface.
+
+```
+   reply(result).header('Last-Modified', lastModified.toUTCString());
+```
+
+There is one more example using `Last-Modified` in [last section](#client-and-server-caching).
+
+### ETag
+
+ETag is alternative to `last-modified` where server provides token (usually checksum of resource) instead of last modified timestamp. Browser in next request sends this ETag in `If-None-Match` header. Server compares it with new `ETag` and return `304` if it's the same.
+
+You only need to set `ETag` in your handler via `etag(tag, options)` function:
+
+```
+   reply(result).etag('xxxxxxxxx');
+```
+
+Check `options` parameter for `etag` function on [api](http://hapijs.com/api#response-object) page for more details.
+
+
+## Server side caching
+
+hapi provides powerful server side caching via [catbox](https://www.github.com/hapijs/catbox) and makes using cache very convenient. This tutorial help you to understand how catbox is utilized inside hapi and how you can use it. Catbox also can be used out of the hapi, but that is not covered here.
+
+Catbox has two interfaces.
+
+
+### Client
+[Client](https://github.com/hapijs/catbox#client) is low-level interface, which basically allows you set/get key-value pairs. Its initialized with one of the available adapters ([Memory](https://github.com/hapijs/catbox-memory), [Redis](https://github.com/hapijs/catbox-redis), [mongoDB](https://github.com/hapijs/catbox-mongodb), [Memcached](https://github.com/hapijs/catbox-memcached), [Riak](https://github.com/DanielBarnes/catbox-riak), [Amazon S3](https://github.com/fhemberger/catbox-s3)).
+
+hapi always initialize one default [client](https://github.com/hapijs/catbox#client) with [memory](https://github.com/hapijs/catbox-memory) adapter. Lets see how we can define more clients.
+
+```javascript
+var Hapi = require('hapi');
+
+var server = new Hapi.Server({
+    cache: [
+        {
+            name: 'mongoCache',
+            engine: require('catbox-mongodb'),
+            host: '127.0.0.1',
+            partition: 'cache',
+        },
+        {
+            name: 'redisCache',
+            engine: require('catbox-redis'),
+            host: '127.0.0.1',
+            partition: 'cache'
+       }
+    ]
+});
+
+server.connection({
+    port: 8000
+});
+```
+
+In this example two [clients](https://github.com/hapijs/catbox#client) were defined: `'mongoCache'`, `'redisCache'` and also default one without name with [memory](https://github.com/hapijs/catbox-memory) adapter. You can replace default client by omitting `name` property. `partition` property tells adapter how cache should be named (`'catbox'` by default). In case of [mongoDB](http://www.mongodb.org/) it is database name and in case of [redis](http://redis.io/) it is used as key prefix.
```

Comment:
Maybe reword to something like:
In Listing 2, we've defined two catbox clients; mongoCache and redisCache. Including the default memory cache created by hapi, there are three available cache clients.